packages/predicates/src/plugins/ownership-predicate.ts

```diff
@@ -0,0 +1,101 @@
+import BigNum = require('bn.js')
+
+import {
+  getOverlappingRange,
+  PredicatePlugin,
+  isRangeSubset,
+  StateUpdate,
+  SyncManager,
+  Transaction,
+} from '@pigi/core'
+
+export class OwnershipPredicatePlugin implements PredicatePlugin {
+  private readonly syncManager: SyncManager
+
+  public constructor(syncManager: SyncManager) {
+    this.syncManager = syncManager
+  }
+
+  public async executeStateTransition(
+    previousStateUpdate: StateUpdate,
+    transaction: Transaction,
+    witness: string
+  ): Promise<StateUpdate> {
+    await this.validateStateTransition(
+      previousStateUpdate,
+      transaction,
+      witness
+    )
+
+    return {
+      range: transaction.range,
+      stateObject: transaction.parameters.newState,
+      depositAddress: transaction.depositAddress,
+      plasmaBlockNumber: transaction.parameters.originBlock,
```

Comment:
The ``plasmaBlockNumber`` for the output of the state transition should always be the``inBlock``--because "that is the block we are executing the transaction in."  

For color, the ``originBlock`` is for the safety check run below, to prevent transaction "replay"--if Alice signs a send to Bob from ``targetBlock==0`` and ``inBlock==1``, and then Bob sends the  coin back to her with ``targetBlock==1`` and ``inBlock==2``, the check guarantees that her previous transaction could not be re-applied because the only valid sends from that new state update would have to have ``originBlock >= 2``.

It's a little confusing--realistically, literally every single predicate plugin will end up returning a state update with ``plasmaBlockNumber === inBlock``.  I thought in the past about having some sort of type which is a state update without the block, but that seemed weird too.  Maybe in reality the predicate plugin only needs to return a ``StateObject``?  And then the state manager would reconstruct the state update based on ``inBlock``, ``transaction.depositAddress``, ``transaction.range``, and the ``stateObject`` output by the ``executeStateTransition``.  Hmmmm... that would definitely work here for ownership, but I'm not sure if it's future-proof.  Paging @karlfloersch for thoughts on how this would work with swaps and whether just returning a state object is sufficient.

Reply:
I like that idea a lot to reduce duplication. Is there ever a case where range, block, or depositAddress would change? I can't think of any, but I could be missing something.

Reply:
Going to leave it as returning a `StateUpdate` for now, but happy to change it if we reach consensus.

Reply:
Sorry, been in transit to Akhila--just confirmed with Karl, he also thinks that using a state object is cleaner and fine for now 👍

Reply:
Since `StateObject` is the only field being mutilated, makes sense for `executeStateTransition` to output a `StateObject` rather than a `StateUpdate`. Thanks for the explanation @ben-chain

Reply:
This is updated now 👍